elements/tcpudp/shifter.cc

```diff
@@ -66,18 +66,19 @@ inline void Shifter::process(Packet *p) {
         q->rewrite_ipport(htonl(ip_src), htons(sport), 0, ip_proto == IP_PROTO_TCP);
         q->rewrite_ipport(htonl(ip_dst), htons(dport), 1, ip_proto == IP_PROTO_TCP);
     }
+    return q;
 }
 
 #if HAVE_BATCH
 void Shifter::push_batch(int, PacketBatch *batch) {
-    FOR_EACH_PACKET(batch, p) { process(p); }
+    EXECUTE_FOR_EACH_PACKET(process,batch);
     output_push_batch(0, batch);
 }
 #endif
 
 Packet *Shifter::simple_action(Packet *p) {
-    process(p);
-    output_push(0, p);
```

Comment:
Also simple_action is a special function, you can't push in that as the system will push the packet returned by the function. So here you would actually push 2 packets out for every packet in, leaking.